linux/keyman-system-service/src/dbus-service.c

```diff
@@ -0,0 +1,121 @@
+// This file implements a service which will run on the system dbus,
+// based on the sd-bus library, see
+// https://0pointer.net/blog/the-new-sd-bus-api-of-systemd.html
+
+#include <errno.h>
+#include <syslog.h>
+#include <systemd/sd-bus.h>
+#include "ServerMethods.h"
+
+#define KEYMAN_BUS_NAME       "com.keyman.SystemService1"
+#define KEYMAN_INTERFACE_NAME "com.keyman.SystemService1.System"
+#define KEYMAN_OBJECT_PATH    "/com/keyman/SystemService1/System"
+
+static int32_t
+on_set_caps_lock_indicator(
+  sd_bus_message *msg,
+  void *user_data,
+  sd_bus_error *ret_error
+) {
+  int32_t ret;
+  uint32_t state;
+
+  ret_error = NULL;
```

Comment:
This is a no-op. Did you mean to set the value rather than the pointer?

```suggestion
  *ret_error = NULL;
```